tokio-fs/src/file/open_options.rs

```diff
@@ -0,0 +1,209 @@
+use super::File;
+
+use futures::{Future, Poll};
+
+use std::fs::OpenOptions as StdOpenOptions;
+use std::io;
+#[cfg(unix)]
+use std::os::unix::fs::OpenOptionsExt as UnixOpenOptionsExt;
+#[cfg(windows)]
+use std::os::windows::fs::OpenOptionsExt as WindowsOpenOptionsExt;
+use std::path::Path;
+
+/// Future returned by `OpenOptions::open` and resolves to a `File` instance.
+#[derive(Debug)]
+pub struct OpenOptionsFuture<P> {
```

Comment:
I wonder if, instead of creating a new future, we can reuse the `OpenFuture`. Since `File::open` *should* be equivalent to `OpenOptions::new().open(path)`, this should be possible.

Thoughts?

Reply:
I'd thought about replacing `OpenFuture` and `CreateFuture` with `OpenOptionsFuture`, but didn't as I figured it would be a breaking change. It didn't occur to me that I could just modify `OpenFuture` for use in both places. I'll do that.